packages/sdk/src/types/runtime/assets/assetManager.ts

```diff
@@ -103,13 +165,21 @@ export class AssetManager {
      */
     public loadGltf(groupName: string, uri: string, colliderType: CreateColliderType = 'none'): Promise<AssetGroup> {
         const p = this.loadGltfHelper(groupName, uri, colliderType);
+
         this.registerLoadPromise(p);
         return p;
     }
 
     private async loadGltfHelper(

```

Comment:
>async [](start = 12, length = 5)

Not your code, but I don't think this needs to be async.

Reply:
It would need a rewrite, but that's out of scope for me. It is currently async, because it awaits a payload being sent to client( load gltf), and result coming back, and then processing newly created assets

---
In reply to: [260900217](https://github.com/Microsoft/mixed-reality-extension-sdk/pull/215#discussion_r260900217) [](ancestors = 260900217)

Reply:
ah, I missed the await. Disregard. It does need to be async then.

---
In reply to: [261010573](https://github.com/Microsoft/mixed-reality-extension-sdk/pull/215#discussion_r261010573) [](ancestors = 261010573,260900217)

Reply:
ah, I missed the await. Disregard. It does need to be async then.

---
In reply to: [261010573](https://github.com/Microsoft/mixed-reality-extension-sdk/pull/215#discussion_r261010573) [](ancestors = 261010573,260900217)